pack3d/pack3d_test.go

```diff
@@ -1,40 +1,84 @@
 package pack3d_test
 
 import (
-	"fmt"
 	"testing"
 
 	"github.com/Authentise/pack3d/pack3d"
 )
 
-func TestCoPack(t *testing.T) {
-	config, err := pack3d.ParseConfig("../tests/fixtures/copack.json")
+func testPackingInputFile(t *testing.T, input string) {
+	config, err := pack3d.ParseConfig(input)
 
-	fmt.Printf("%v", config)
 	if err != nil {
 		t.Fatalf("Failed to load config: %s", err)
 	}
 
-	_, err = pack3d.Pack(config)
+	output, err := pack3d.Pack(config)
 
 	if err != nil {
-		t.Fatalf("Failed to pack model: %s", err)
+		t.Fatalf("Failed to pack config: %s", err)
+	}
+
+	if len(output.Model.Items) != config.TotalItems() {
+		t.Fatalf("Failed to pack all models: packed %d of %d", len(output.Model.Items), config.TotalItems())
 	}
 }
 
-// Tests an input file with old master-ricoh api
-// These input files don't have axes_lock and spacing fields
-func TestMasterRicohCoPack(t *testing.T) {
-	config, err := pack3d.ParseConfig("../tests/fixtures/master-ricoh-copack.json")
+// Pass in the expected number of items packed
+func testPartialPackingInputFile(t *testing.T, input string, expectedPacking int) {
+	config, err := pack3d.ParseConfig(input)
 
-	fmt.Printf("%v", config)
 	if err != nil {
 		t.Fatalf("Failed to load config: %s", err)
 	}
 
-	_, err = pack3d.Pack(config)
+	output, err := pack3d.Pack(config)
 
 	if err != nil {
-		t.Fatalf("Failed to pack model: %s", err)
+		t.Fatalf("Failed to pack config: %s", err)
+	}
+
+	if len(output.Model.Items) != expectedPacking {
+		t.Fatalf("Failed to pack all models: packed %d of %d", len(output.Model.Items), expectedPacking)
 	}
 }
+
+func TestCoPack(t *testing.T) {
+	testPackingInputFile(t, "../tests/fixtures/copack.json")
+}
+
+// Tests an input file with old master-ricoh api
+// These input files don't have axes_lock and spacing fields
+func TestMasterRicohCoPack(t *testing.T) {
+	testPackingInputFile(t, "../tests/fixtures/master-ricoh-copack.json")
+}
+
+func TestBuildPlateTooSmall(t *testing.T) {
+	// NOTE: This test fails in master-ricoh but NOT in dev, and doesn't fail here.
+	t.Skip()
+	testPartialPackingInputFile(t, "../tests/fixtures/too-small.json", 1)
+}
+
+func TestPartialPack(t *testing.T) {
+	// NOTE: This test fails in master-ricoh but NOT in dev, and doesn't fail here.
+	t.Skip()
+	testPartialPackingInputFile(t, "../tests/fixtures/partial-pack.json", 2) // Currently 3
+}
+
+func TestSc45665(t *testing.T) {
```

Comment:
Please make human readable / purpose names here.   God knows we change Repos or tracking tools every 2 years. 

Cut a ticket to 'insert real problem cases here' for @francis-authentise to handle, please (sorry Francis !)